src/structure_managers/adaptor_neighbour_list.hh

```diff
@@ -555,24 +559,25 @@ namespace rascal {
       // considering the neighbours of ghosts, it is possible to iterate over
       // all atoms including ghosts using the proxy `.with_ghosts`, which takes
       // its `.end()` from here`
-      auto nb_atoms{this->consider_ghost_neighbours ? this->n_centers + n_ghosts
-                                                    : this->get_size()};
+      auto nb_atoms{this->consider_ghost_neighbours
+                        ? this->n_centers + this->n_ghosts
+                        : this->get_size()};
       return nb_atoms;
     }
 
     //! Returns position of an atom with index atom_tag
     inline Vector_ref get_position(const size_t & atom_tag) {
-      if (atom_tag < n_centers) {
+      if (atom_tag < this->n_atoms) {
```

Comment:
I guess this is needed because of the changed number of center atoms?

Reply:
yes